subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadIntegrationTest.groovy

```diff
@@ -16,14 +16,53 @@
 
 package org.gradle.jvm.toolchain
 
+
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
 import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
-import org.hamcrest.CoreMatchers
-
-import static org.hamcrest.CoreMatchers.containsString
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.test.fixtures.file.TestFile
 
 class JavaToolchainDownloadIntegrationTest extends AbstractIntegrationSpec {
 
+    def "download and provisioning works end-to-end"() {
+        buildFile << """
+            apply plugin: "java"
+
+            java {
+                toolchain {
+                    languageVersion = JavaLanguageVersion.of(8)
+                }
+            }
+        """
+
+        file("src/main/java/Foo.java") << "public class Foo {}"
+
+
+        when:
+        executer
+                .withTasks("compileJava")
+                .requireOwnGradleUserHomeDir()
+                .withToolchainDownloadEnabled()
+                .expectDocumentedDeprecationWarning("Starting from Gradle 8.0 there will be no default Java Toolchain Registry. " +
```

Comment:
I think the message may be a bit unclear: a user is asking for a toolchain, and gets an error about a "Java Toolchain Registry" and the need to "inject such registries". The error message should probably say something more specific, like:

"Your build script declares a Java Toolchain, but no toolchain registry is declared, which is going to be an error in Gradle 8. In order to declare a registry for toolchains, you must edit your settings script and add one via the `toolchainManagement` block".

Reply:
Makes sense, updated the message.